src/app/GitUI/Editor/Diff/DifftasticHighlightService.cs

```diff
@@ -162,92 +147,66 @@ public DifftasticHighlightService(ref string text, DiffViewerLineNumberControl l
+                // Add spaces so both-sides markers are aligned
+                columnGap = rightColumnStart - rightStartOffset;
+                if (columnGap > 0)
                 {
-                    rightLineNo = lineNo;
+                    lineBuilder = lineBuilder.Insert(rightStartOffset, new string(' ', columnGap));
                 }
+            }
 
-                int rightLen = matchRight.Length;
-                lineBuilder = lineBuilder.Remove(rightStartOffset, rightLen);
-
-                int i = 0;
-                bool first = true;
-                while (i < textMarkers.Count)
+            bool first = true;
+            foreach (TextMarker tm in textMarkers)
+            {
+                if (tm.EndOffset < rightStartOffset)
                 {
-                    if (textMarkers[i].EndOffset < rightStartOffset)
-                    {
-                        ++i;
-                        continue;
-                    }
-
-                    if (textMarkers[i].Offset >= rightStartOffset + rightLen)
-                    {
-                        break;
-                    }
-
-                    if (first)
-                    {
-                        first = false;
-
-                        Color c = AppSettings.ReverseGitColoring.Value ? textMarkers[i].Color : textMarkers[i].ForeColor;
-                        if (c.R != c.G)
-                        {
-                            // Merge line type with existing left line type
-                            lineType = lineType == DiffLineType.Context ? DiffLineType.PlusRight : DiffLineType.MinusPlus;
-                        }
-                    }
-
-                    if (textMarkers[i].Offset > rightStartOffset)
-                    {
-                        textMarkers[i].Length -= textMarkers[i].Offset - rightStartOffset;
-                        textMarkers[i].Offset = rightStartOffset;
-                    }
-
-                    if (textMarkers[i].EndOffset < rightStartOffset + rightLen - 1)
-                    {
-                        // textMarkers[i].Length += rightStartOffset + rightLen - textMarkers[i].EndOffset;
-                    }
-
-                    if (textMarkers[i].Length <= 0
-                        || (rightStartOffset >= textMarkers[i].Offset && textMarkers[i].EndOffset <= rightStartOffset + rightLen))
-                    {
-                        textMarkers.RemoveAt(i);
-                        --i;
-                    }
-
-                    ++i;
+                    continue;
                 }
 
-                if (rightStartOffset > 0)
+                if (first)
                 {
-                    if (rightColumnStart == 0)
+                    first = false;
+
+                    Color c = AppSettings.ReverseGitColoring.Value ? tm.Color : tm.ForeColor;
+                    if (c.R != c.G)
```

Comment:
I guess this selects (more or less) red or green lines but not neutral colors. A comment might help future readers.

Reply:
Tried to clarify the heuristics here and on line 93